src/media/js/elements/nav.js

```diff
@@ -0,0 +1,149 @@
+            subnavs: {
+                get: function() {
+                    return $(this).children('mkt-nav-child');
+                },
+            },
+        }),
+    });
+
+    var MktNavRootElement = document.registerElement('mkt-nav-root', {
+        prototype: Object.create(HTMLUListElement.prototype, {}),
+    });
+
+    var MktNavChildElement = document.registerElement('mkt-nav-child', {
+        prototype: Object.create(MktNavRootElement.prototype, {
+
+            // Show this subnav and hide all sibling <mkt-nav-child> elements.
+            show: {
+                value: function() {
+                    var $this = $(this);
+                    $this.siblings('.' + VISIBLE).removeClass(VISIBLE);
+                    $this.addClass(VISIBLE);
+                    return this;
+                },
+            }
+        }),
+    });
+
+    // Toggle the <mkt-nav> element with the specified id when elements with the
+    // data-toggle-nav attribute are clicked.
+    z.body.on('click', '[data-toggle-nav]', function(evt) {
+        evt.preventDefault();
+        evt.stopPropagation();
+        var navId = $(this).data('toggle-nav');
+        $('#' + navId).get(0).toggle();
+    })
+
+    // Toggle the <mkt-nav-child> element with the specified id when elements
+    // with the data-toggle-nav attribute are clicked. If the value of that
+    // attribute is empty, attempt to toggle the parent's subnav if it is an
+    // <mkt-nav> element.
+    .on('click', '[data-toggle-subnav]', function(evt) {
+        evt.preventDefault();
+        evt.stopPropagation();
+        var $this = $(this);
+        var id = $this.data('toggle-subnav');
+        if (id) {
+            var $subnav = $('#' + id);
+            if ($subnav.is('.' + VISIBLE)) {
+                $subnav.parent().get(0).toggleSubnav();
+                $this.removeClass(SHOWING);
+            } else {
+                $subnav.get(0).show();
+                $subnav.parent().get(0).toggle(true).toggleSubnav(true);
+                $this.addClass(SHOWING);
+            }
+        } else {
+            var $parent = $this.parent();
+            if ($parent.is('mkt-nav')){
+                $parent.get(0).toggleSubnav();
+            }
+        }
+    });
+
+    $(function() {
+
+    })
+
+    z.page.on('navigate', function(evt) {
+        // Remove highlights from formerly-active nodes.
+        $('mkt-nav a.' + ACTIVE).each(function(index, element) {
+            $(element).removeClass(ACTIVE);
+        });
+
+        // Highlight new active nodes.
+        $('mkt-nav a[href="' + window.location.pathname + '"]').each(function(index, element) {
+            $(element).addClass(ACTIVE);
+        });
+
+        // Close all menus.
+        $('mkt-nav').each(function(index, element) {
+            if (element.hasOwnProperty('toggle')) {
+                element.toggle(false).toggleSubnav(false); 
+            }
+        });
+
+        $('.' + SHOWING).removeClass(SHOWING);
```

Comment:
I understand the sentiment, though I think it's fine to mix use of jQuery with the use of native DOM API. We do it everywhere, and code is code. Using jQuery when we don't have to, in these cases, is a perf hit.